src/material-experimental/mdc-tabs/ink-bar.ts

```diff
@@ -128,16 +141,36 @@ export class MatInkBarFoundation {
     this._destroyed = true;
   }
 
+  /** Creates and appends the indicator element. */
   private _createIndicator(document: Document) {
+    this._indicator = document.createElement('span');
```

Comment:
The name of this class seems a little strange, it's easy to confuse with the concept of a `Foundation` in MDC. (I know it wasn't created in this PR, but just something I noticed now).

Why does this class do so much direct DOM manipulation as opposed to being its own component with its own template?

Reply:
I called this class a foundation because it's a wrapper around MDC's `MDCSlidingTabIndicatorFoundation`.

The reason why it needs to do so much manual DOM manipulation is because we use this both for the tab group and the tab nav bar. The nav bar uses a directive for the links so we don't have control over the template.

Reply:
Ah, I see. Obviously outside the scope of this PR, but maybe we could encapsulate the behavior in a component and then just use `ViewContainerRef` to create it and attach it to the element with the directive? Not sure if that's actually feasible, just brainstorming...

Reply:
I think the boilerplate-wise it would end up being about the same amount of code. Ideally we'd turn the tab link into a component, but that would be a breaking change.